src/charm.py

```diff
@@ -80,6 +203,23 @@ def _on_leader_elected(self, _) -> None:
         if replication_password is None:
             self._peers.data[self.app]["replication-password"] = self._new_password()
 
+        # Create resources and add labels needed for replication.
+        self._create_resources()
+
+        # Patch the pod labels of this unit to enable replication later.
+        self._patch_pod_labels(self.unit.name)
```

Comment:
If we keep it in on install it will add the replicas to the cluster too soon (in a unknown state that prevents us from checking later that all members are ready; we can create some logic to handle this unknown state in the `are_all_members_ready` function but it could generate some confusion for someone that is reading the code). So, it's better to add the pod labels later. I moved it to both pebble ready and add cluster member functions on https://github.com/canonical/postgresql-k8s-operator/pull/9/commits/ee0018899c32c6f9b60debcc5179ef39eb6ac02c, so it's done only once per unit.